components/tikv_util/src/deadline.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use super::time::{Duration, Instant};
+
+#[derive(Debug, Copy, Clone)]
+pub struct DeadlineError;
```

Comment:
Why not derive from `Fail` which can implement `description` automatically.

Reply:
Currently there is no fail in tikv_util and I don't want to import it for this simple purpose.